### ISSUE TITLE
Bug with MAML's value function training?

### ISSUE THREAD

**Issue by YeeCY** [rlworkgroup/garage#2301]
As far as I know, the current code of MAML only fits the value function to episodes from one ground of on-policy rollouts during adaptation. But the code here seems to have a bug:

https://github.com/rlworkgroup/garage/blob/731101898450da6c64fd6f6cdfc9910e4a7d8ea6/src/garage/torch/algos/maml.py#L176

This loss only backpropagates gradients into `self._value_function` which is a deep copy of `inner_algo._value_function`:

https://github.com/rlworkgroup/garage/blob/731101898450da6c64fd6f6cdfc9910e4a7d8ea6/src/garage/torch/algos/maml.py#L67

However, the optimizer is actually optimizing `inner_algo._value_function` instead of `self._value_function`:

https://github.com/rlworkgroup/garage/blob/731101898450da6c64fd6f6cdfc9910e4a7d8ea6/src/garage/torch/algos/maml.py#L178

It seems the value function never gets updated. I can confirm that there are gradients for `self._value_function` but no gradients for `inner_algo._value_function`:

![image](https://user-images.githubusercontent.com/40682795/132279444-eab9fbc0-a034-4b7d-ac74-8a4ed1d59b97.png)

**Comment by avnishn** [comment]
to use a working version of MAML, check out #2287
